crates/sui-transaction-builder/src/lib.rs

```diff
@@ -0,0 +1,1046 @@
+    /// Set the expiration of the transaction.
+    pub fn set_expiration(&mut self, epoch: u64) {
+        self.expiration = Some(TransactionExpiration::Epoch(epoch));
+    }
+
+    // Commands
+    /// Call a public or a Move function with the given arguments.
+    ///
+    /// - `function` is a structured representation of a package::module::function argument.
+    ///
+    /// The return value is a result argument that can be used in subsequent commands.
+    /// If the move call returns multiple results, you can access them using the
+    /// [`Argument::nested`] method.
+    pub fn move_call(&mut self, function: Function, arguments: Vec<Argument>) -> Argument {
+        let cmd = Command::MoveCall(MoveCall {
+            package: function.package.into(),
+            module: function.module,
+            function: function.function,
+            type_arguments: function.type_args,
+            arguments,
+        });
+        self.commands.push(cmd);
+        Argument::Result(self.commands.len() as u16 - 1)
+    }
+
+    /// Transfer a list of objects to the given address, without producing any result.
+    pub fn transfer_objects(&mut self, objects: Vec<Argument>, address: Argument) {
+        let cmd = Command::TransferObjects(TransferObjects {
+            objects: objects.into_iter().collect(),
+            address,
+        });
+        self.commands.push(cmd);
+    }
+
+    /// Splits a coin by the provided amounts, returning multiple results (as many as there are
+    /// amounts. To access the results, use the [`Argument::nested`] method to access the desired
+    /// coin by its index.
+    pub fn split_coins(&mut self, coin: Argument, amounts: Vec<Argument>) -> Argument {
+        let cmd = Command::SplitCoins(SplitCoins {
+            coin,
+            amounts: amounts.into_iter().collect(),
+        });
+        self.commands.push(cmd);
+        Argument::Result(self.commands.len() as u16 - 1)
+    }
+
+    /// Merge a list of coins into a single coin, without producing any result.
+    pub fn merge_coins(&mut self, into_coin: Argument, coins: Vec<Argument>) {
+        let cmd = Command::MergeCoins(MergeCoins {
+            coin: into_coin,
+            coins_to_merge: coins.into_iter().collect(),
+        });
+        self.commands.push(cmd);
+    }
+
+    /// Make a move vector from a list of elements. If the elements are not objects, or the vector
+    /// is empty, a type must be supplied.
+    /// It returns the Move vector as an argument, that can be used in subsequent commands.
+    pub fn make_move_vec(&mut self, type_: Option<TypeTag>, elements: Vec<Argument>) -> Argument {
+        let cmd = Command::MakeMoveVector(MakeMoveVector {
+            type_,
+            elements: elements.into_iter().collect(),
+        });
+        self.commands.push(cmd);
+        Argument::Result(self.commands.len() as u16 - 1)
+    }
+
+    /// Publish a list of modules with the given dependencies. The result is the upgrade cap, which
```

Comment:
```suggestion
    /// Publish a list of modules with the given dependencies. The result is the `0x2::package::UpgradeCap`, which
```

Reply:
Reword this to mention that you need to process the upgrade cap: burn it, send it, wrap it.